### PR TITLE
render from a string

### LINKED ISSUE #22: Allow rendering from a string
Instead of rendering a file, render a string

### PR DESCRIPTION
**Pull request by Stephn-R**
Closes #22

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -8,9 +8,18 @@ var tagParser = require('./app/tag_parser');
...
 
 var xejs = function(file, options, parentPath) {
-    var dirname = file;
-    if (parentPath) dirname = path.join(parentPath, "../", file);
-    var content = fs.readFileSync(dirname, 'utf-8');
+    var content;
+
+    if(options.stringRender) {
+        // Render contents from file var
+        content = file;
```
Comment (angrykoala) [inline index.js]: What happens with included files when rendering from a string?

Comment (Stephn-R) [inline index.js]: Question: @angrykoala , is it expected that the string rendering should apply to nested includes as well? If so, then we would not resolve include statements

Comment (angrykoala) [inline index.js]: I think not applying the string render to includes would be the best thing to do. what do you think?

Comment (angrykoala) [inline index.js]: Sorry, I was thinking of doing the exact opposite. Even from a string, I would expect the same behavior and get the tags (and includes) resolved

Comment (Stephn-R) [inline index.js]: That is the fact currently. It does resolve includes and tags currently.

Comment (Stephn-R) [inline index.js]: @angrykoala the branch has been changed

File: test/main.js [hunk 2]
```diff
@@ -59,4 +59,18 @@ describe("Main test", function() {
             });
         });
     });
+
+    it('Render from a string', function(done) {
+        var fileContents = fs.readFileSync(__dirname + '/file1.md', 'utf-8');
+
+        xejs(fileContents, { stringRender: true }, function(err, res) {
```
Comment (angrykoala) [inline test/main.js]: Here we should test that the result is what we want

Comment (angrykoala) [inline test/main.js]: There are conflicts in index.js and test/main.js that have to be fixed before merge

**Comment by angrykoala** [comment]
Thanks for your contribution @Stephn-R, however due to several changes in the core code of _xejs_ and some changes in the interface for string rendering I could not merge your changes into the current solution for #22 so I will close this PR (and soon the issue #22)
Thanks again for your help developing this issue

**Pull request closed without merging**